apps/console/src/features/roles/components/edit-role/edit-role-basic.tsx

```diff
@@ -195,42 +77,77 @@ export const BasicRoleDetails: FunctionComponent<BasicRoleProps> = (props: Basic
      *
      * @param id - Role ID which needs to be deleted
      */
-    const handleOnDelete = (id: string): void => {
-        deleteRoleById(id).then(() => {
+    const handleOnDelete = (): void => {
+        deleteRoleById(role.id).then(() => {
             handleAlerts({
                 description: t("console:manage.features.roles.notifications.deleteRole.success.description"),
                 level: AlertLevels.SUCCESS,
                 message: t("console:manage.features.roles.notifications.deleteRole.success.message")
             });
-            if (isGroup) {
-                history.push(AppConstants.getPaths().get("GROUPS"));
-            } else {
-                history.push(AppConstants.getPaths().get("ROLES"));
-            }
+
+            history.push(AppConstants.getPaths().get("ROLES"));
         });
     };
 
     /**
-     * Method to update role name for the selected role.
+     * Validates the Form.
      *
+     * @param values - Form Values.
+     * @returns Form validation.
+     */
+    const validateForm = async (values: RoleBasicInterface): Promise<RoleBasicInterface> => {
+
+        const errors: RoleBasicInterface = {
+            roleName: undefined
+        };
+
+        if (values.roleName?.toString().trim().length >= 3) {
+
+            const roleName: string = values.roleName?.toString().trim();
+
+            try {
+                setRoleNameSearchQuery(`displayName eq ${roleName}`);
+
+                if (!isRolesListLoading || !isRolesListValidating) {
+                    if (rolesList?.totalResults > 0) {
+                        errors.roleName = t("console:manage.features.roles.addRoleWizard.forms."
+                            + "roleBasicDetails.roleName.validations.duplicate",{ type: "Role" });
+                    }
+                } 
+            } catch (error: unknown) {
```

Comment:
Shall we add data type?

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4326/commits/317543c8be8c0461214e6e67425f492c835d6bd7